jax/_src/lax/control_flow/common.py

```diff
@@ -58,17 +57,24 @@ def _initial_style_open_jaxpr(fun: Callable, in_tree, in_avals,
   wrapped_fun, out_tree = flatten_fun_nokwargs(lu.wrap_init(fun), in_tree)
   debug = pe.debug_info(fun, in_tree, out_tree, False,
                         primitive_name or "<unknown>")
-  jaxpr, _, consts, () = pe.trace_to_jaxpr_dynamic(wrapped_fun, in_avals, debug)
-  return jaxpr, consts, out_tree()
+  jaxpr, _, consts, attrs_tracked = pe.trace_to_jaxpr_dynamic(
+      wrapped_fun, in_avals, debug)
+  return jaxpr, consts, out_tree(), attrs_tracked
 
-@weakref_lru_cache
```

Comment:
This change isn't strictly related to this PR.

This cache is redundant because we already cache `_initial_style_open_jaxpr`, `pe.convert_constvars_jaxpr`, and `pe.close_jaxpr`, each of the three functions that this function applies. I removed the decorator on this function just to cut down on redundant caches.